pypeit/spectrographs/keck_deimos.py

```diff
@@ -426,6 +439,255 @@ def setup_arcparam(self, arcparam, disperser=None, fitstbl=None, arc_idx=None,
         else:
             msgs.error('Not ready for this disperser {:s}!'.format(disperser))
 
+    def get_slitmask(self, filename):
+        hdu = fits.open(filename)
+        corners = np.array([hdu['BluSlits'].data['slitX1'],
+                            hdu['BluSlits'].data['slitY1'],
+                            hdu['BluSlits'].data['slitX2'],
+                            hdu['BluSlits'].data['slitY2'],
+                            hdu['BluSlits'].data['slitX3'],
+                            hdu['BluSlits'].data['slitY3'],
+                            hdu['BluSlits'].data['slitX4'],
+                            hdu['BluSlits'].data['slitY4']]).T.reshape(-1,4,2)
+        self.slitmask = SlitMask(corners, slitid=hdu['BluSlits'].data['dSlitId'])
+        return self.slitmask
+
+    def get_grating(self, filename):
+        """
+        Take from xidl/DEEP2/spec2d/pro/deimos_omodel.pro and
+        xidl/DEEP2/spec2d/pro/deimos_grating.pro
+        """
+        hdu = fits.open(filename)
+
+        # Grating slider
+        slider = hdu[0].header['GRATEPOS']
+        # TODO: Add test for slider
+
+        # Central wavelength, grating angle, and tilt position
+        if slider == 3:
+            central_wave = hdu[0].header['G3TLTWAV']
+            # Not used
+            #angle = (hdu[0].header['G3TLTRAW'] + 29094)/2500
+            tilt = hdu[0].header['G3TLTVAL']
+        else:
```

Comment:
should we make this 

``` 
elif slider in [2,4]:
```

and than have an `else` catching any unexpected value?

Reply:
Done.  Does anyone know if there are more than these three positions?